pkg/exasol-rest-api/websocket_connection.go

```diff
@@ -0,0 +1,183 @@
+package exasol_rest_api
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"database/sql/driver"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/websocket"
+	"math/big"
+	"net/url"
+	"os/user"
+	"runtime"
+	"strconv"
+)
+
+type websocketConnection struct {
+	connProperties *ApplicationProperties
+	websocket      *websocket.Conn
+}
+
+func (connection *websocketConnection) close() error {
+	err := connection.send(&Command{Command: "disconnect"}, nil)
+	connection.websocket.Close()
+	connection.websocket = nil
+	return err
+}
+
+func (connection *websocketConnection) connect() error {
```

Comment:
Move this function before `close`, because you connect *before* closing the connection

Reply:
Moved!